### PR TITLE
Bump to 1.18.0 and add Windows support

### LINKED ISSUE #41: Begin using meson build system
Currently GStreamer 1.16 is being built, which supports both meson and automake, however in master and for the next 1.18 release, automake has been removed completely in support of meson. Some feedstock that have moved to meson include [atk](https://github.com/conda-forge/atk-feedstock) and [gdk-pixbuf](https://github.com/conda-forge/gdk-pixbuf-feedstock).

I'll take a stab at doing this, and submit a PR if successful

### PR DESCRIPTION
**Pull request by tschoonj**
Closes #41

### CODE REVIEW

**Comment by tschoonj** [comment]
By the way, I recommend that gst-python, which currently has its own [feedstock](https://github.com/conda-forge/gst-python-feedstock), is also made part of this recipe, given that all gstreamer packages are released simultaneously.

**Comment by tschoonj** [comment]
@msarahan @ccordoba12 @mingwandroid 

Could someone review this PR please? Thanks!

**Comment by tschoonj** [comment]
@isuruf @pkgw @xhochy @scopatz 

Could you comment on this PR please?

Many thanks in advance!

**Review by pkgw — COMMENTED** [review]
Nice work! This looks very thorough.

I'm not familiar with the details of this package. I noticed two things:

First, adding the "build" prefix to the pkg-config search path is a bit of a red flag, which might indicate future problems cross-compiling. Related are these messages during the build:

```
Warning: rpath {BLAH}/_build_env/lib is outside prefix {PREFIX} (removing it)
```

But if the rpath is being removed and the tests are passing, it might all be OK.

Second, I see these sorts of messages being printed during the test runs:

```
+ gst-typefind-1.0 --version
gst-typefind-1.0 version 1.18.1
GStreamer 1.18.1
Unknown package origin
```

Does this mean that GStreamer has some kind of prefix-detection code that isn't working right when conda-ified?

**Comment by tschoonj** [comment]
> Nice work! This looks very thorough.
> 

Thanks!

> I'm not familiar with the details of this package. I noticed two things:
> 
> First, adding the "build" prefix to the pkg-config search path is a bit of a red flag, which might indicate future problems cross-compiling. Related are these messages during the build:
> 
> ```
> Warning: rpath {BLAH}/_build_env/lib is outside prefix {PREFIX} (removing it)
> ```
> 
> But if the rpath is being removed and the tests are passing, it might all be OK.
> 

This is due to gobject-introspection and meson not playing nice together if they are not together in `build`. I have done something similar for other GNOME projects as well.

> Second, I see these sorts of messages being printed during the test runs:
> 
> ```
> + gst-typefind-1.0 --version
> gst-typefind-1.0 version 1.18.1
> GStreamer 1.18.1
> Unknown package origin
> ```
> 
> Does this mean that GStreamer has some kind of prefix-detection code that isn't working right when conda-ified?

This looks normal to me. I get the exact same output when using GStreamer from Homebrew.

**Comment by pkgw** [comment]
> This is due to gobject-introspection and meson not playing nice together if they are not together in `build`. I have done something similar for other GNOME projects as well.

Ah, OK. "Someone" really ought to split up the gobject-introspection package — but unfortunately this isn't a task that I can realistically take on any time soon :-(

File: recipe/meta.yaml [hunk 1]
```diff
@@ -168,7 +192,7 @@ outputs:
 about:
   home: http://gstreamer.freedesktop.org/
```
Comment (Anthchirp) [inline recipe/meta.yaml:193]:
```suggestion
  home: https://gstreamer.freedesktop.org/
```

Comment (tschoonj) [reply recipe/meta.yaml]: Thanks Markus!

**Comment by pkgw** [comment]
Anyone else have any feedback? I will merge in a few days if not.

**Comment by tschoonj** [comment]
Any thoughts on me adding gst-python to the recipe?

**Comment by pkgw** [comment]
@tschoonj Thanks for this PR! I would support adding gst-python, but (evidently) in a separate PR to keep the review scope in bounds.

**Pull request merged**